src/components/pagination/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { useMemo, useState } from 'react'
+import {
+  StandaloneChevronLeftRegularIcon,
+  StandaloneChevronRightRegularIcon,
+} from '@deriv/quill-icons'
+import { usePaginationRange } from 'hooks/usePaginationRange'
+import PaginationButton from './pagination-button'
+import {
+  navigationIconFillVariants,
+  paginationVariants,
+} from './pagination.classnames'
+import { PaginationProps } from './pagination.types'
+
+/**
+ * Component that can divide large lists of data into smaller more manageable pages.
+ * @name Pagination
+ * @param {string} className- Styles to be applied to Pagination body
+ * @param {number} contentPerPage- Number of items to be displayed per page
+ * @param {Array} dataList- List of data to be displayed
+ * @param {React.Component} renderComponent- Component to be rendered
+ * @param {string} variant- Variant of Pagination.
+ *
+ * @example
+ * <Pagination
+ * className="pagination__container"
+ * contentPerPage={10}
+ * dataList={dataList}
+ * renderComponent={<DisplayContent />}
+ * variant="number"
+ * />
+ */
+const Pagination = <T,>({
+  className,
+  contentPerPage,
+  dataList = [],
+  renderComponent,
+  variant = 'number',
+}: PaginationProps<T>) => {
+  const [currentPage, setCurrentPage] = useState(1)
+
+  const totalPageCount = useMemo(() => {
+    const dataToDisplay = contentPerPage > 0 ? contentPerPage : 1
+    return Math.ceil(dataList.length / dataToDisplay)
+  }, [dataList.length, contentPerPage])
+
+  const RenderComponent = renderComponent
+
+  const paginationRange = usePaginationRange({
+    totalPageCount,
+    currentPage,
+    variant,
+  })
+
+  const goToNextPage = () => setCurrentPage((page) => page + 1)
+
+  const gotToPreviousPage = () => setCurrentPage((page) => page - 1)
+
+  const changePage = (event: string | null) => {
+    const pageNumber = Number(event)
+    setCurrentPage(pageNumber)
+  }
+
+  const getPaginatedData = () => {
+    const startIndex = currentPage * contentPerPage - contentPerPage
+    const endIndex = startIndex + contentPerPage
+    return dataList.slice(startIndex, endIndex)
+  }
+
+  return (
+    <div className="row-auto grid">
+      <section className={className}>
+        {getPaginatedData().map((dataItem) => (
+          <RenderComponent key={dataItem.id} data={dataItem} />
+        ))}
```

Comment:
Why is it necessary to include this component? It should function as a standalone component. You can refer to the documentation [here](https://mui.com/material-ui/react-pagination/). It exclusively focuses on the pagination component and doesn't depend on the List component.

Reply:
I've removed the prop that accepts a component